### ISSUE TITLE
history/get_actions only returns initiated actions for an account. Is there a way to get recieved actions for an account?

### ISSUE THREAD

**Issue by ismyhc** [EOSIO/eos#4274]
Hi,

Im trying to get a list of transactions/action for an account. I am able to use the history/get_actions rpc interface but that seems only give initiated/sent actions for the account. How can I get received actions/transactions for an account through the rpc interface?

Any help much appreciated :)

**Comment by liliyongleo** [comment]
same problem.......    it could give send/receive actions in previous version , e.g. dawn 4.1...

Please help. Thanks.

**Comment by ismyhc** [comment]
@liliyongleo Ah, okay. So get_actions was returning send/receive actions at one point?

**Comment by ismyhc** [comment]
@andriantolie So to confirm, get_actions will only return actions that where initiated by the account passed in?

**Comment by andriantolie** [comment]
Nope, you can also get it using get_actions. It will return all the actions both received and signed by the filtered account. For example, if you call `cleos get actions eosio.token` and you enable `filter-on = eosio.token:transfer:`, you will get all list of users who have done a transfer. If you want to have the actions executed before you enabled your history_api_plugin to get tracked, you will need to replay your blockchain though.

**Comment by wuyahuang** [comment]
The issue is if you try to get actions by account name, then you need to put `filter-on = ACCOUNT_NAME:transfer:` to config.ini and replay, but it's impossible to put every account name to config.ini.

**Comment by dyba** [comment]
I have a similar problem. I'm trying to retrieve `Received` actions. In particular, I noticed that when I tried to read `Received` actions for an account, they didn't show up when calling the `getActions` API call. I'm using EOS JS API: https://github.com/EOSIO/eosjs-api. I've read through the EOS JS API code and notice that it is a very thin client wrapper for EOS. Are there plans for EOS to support reading `Received` actions when retrieving actions tied to an EOS account?
